api/routes/auth.js

```diff
@@ -0,0 +1,135 @@
+const router = require('express').Router()
+const User = require('../models/user');
+const bcrypt = require('bcrypt');
+const jsonwebtoken = require('jsonwebtoken');
+
+const {SECRET_KEY} = process.env
+
+const saltRounds = 10;
+
+// Add a post route for sign up
+router.post('/signup', async (req, res, next) =>{
+    var status = 201;
+    try {
+        const {username, password} = req.body;
+        if (!username)
+        {
+            status = 400;
+            throw new Error('Please provide a valid username');
+        }
+        if (!password)
+        {
+            status = 400;
+            throw new Error('Please provide a valid password');
+        }
+        if(password.length < 8)
+        {
+            status = 400;
+            throw new Error('Your password should be atleast 8 characters');
+        }
+        var user = await User.findOne({username});
+        if (user)
+        {
+            status = 400;
+            throw new Error('User already exists');
+        }
+        var hashedPassword = await bcrypt.hash(req.body.password, saltRounds);
+        const response = await User.create({username : username, password: hashedPassword});
+        res.status(status).json({status, response})
+    }
+    catch(err){
+        console.error(err);
+        const error  = new Error(err.message)
+        error.status = 400
+        next(error)
+    }
+})
+
+// Add a post route for login
+router.post('/login', async (req, res, next) => {
+    var status = 201;
+
+    try{
+        const {username, password} = req.body;
+        var guest = await User.findOne({username})
+        if (!guest)
+        {
+           throw new Error('User not found')
+        }
+
+        const isValid = await bcrypt.compare(password, guest.password)
+        if (!isValid) throw new Error('Password is invalid');
```

Comment:
You'll want to use the same error message for both of the above errors. Otherwise, it's easier for someone to "guess" another user's username and is more insecure.